### PR TITLE
Do pod preemption in parallel.

### LINKED ISSUE #113007: Do pod preemption in parallel
### What would you like to be added?

When a pod requires preempting more than one pod, this happens sequentially. It should be done in parallel. https://github.com/kubernetes/kubernetes/blob/525280d285c4bb4970e571a1e13a601befd75434/pkg/scheduler/framework/preemption/preemption.go#L338-L370

/sig scheduling
/sig scalability

### Why is this needed?

Preemption happens in the scheduling cycle, meaning that no other scheduling attempts happen while preemption is ongoing, hurting throughput.

An alternative is to the deletes in a separate routine, like we do for binding.

### PR DESCRIPTION
**Pull request by tangwz**
#### What type of PR is this?

/sig scheduling
/sig scalability
/kind cleanup

#### What this PR does / why we need it:

Preemption happens in the scheduling cycle, meaning that no other scheduling attempts happen while preemption is ongoing, hurting throughput.
When a pod requires preempting more than one pod, this happens sequentially. It should be done in parallel.

#### Which issue(s) this PR fixes:

Fixes #113007

#### Special notes for your reviewer:
@alculquicondor

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by sanposhiho — COMMENTED** [review]
/ok-to-test

**Review by sanposhiho — APPROVED** [review]
Looks good.
/lgtm

File: pkg/scheduler/framework/preemption/preemption.go [hunk 1]
```diff
@@ -355,18 +359,26 @@ func (ev *Evaluator) prepareCandidate(ctx context.Context, c Candidate, pod *v1.
 				if apipod.UpdatePodCondition(newStatus, condition) {
 					if err := util.PatchPodStatus(ctx, cs, victim, newStatus); err != nil {
 						klog.ErrorS(err, "Preparing pod preemption", "pod", klog.KObj(victim), "preemptor", klog.KObj(pod))
-						return framework.AsStatus(err)
+						errCh.SendError(err)
```
Comment (sanposhiho) [inline pkg/scheduler/framework/preemption/preemption.go:362]: `SendErrorWithCancel` might be better.

**Review by sanposhiho — COMMENTED** [review]
/remove-lgtm

Sorry, one nits.

File: pkg/scheduler/framework/preemption/preemption.go (hunk 1 above)
Comment (sanposhiho) [reply pkg/scheduler/framework/preemption/preemption.go:362]: One example of `SendErrorWithCancel` you can refer is ↓
https://github.com/kubernetes/kubernetes/blob/fd358d360618bcc30ed9579a9cfd25c7fdc293f4/pkg/scheduler/framework/runtime/framework.go#L900

**Comment by tangwz** [comment]
@sanposhiho PTLA. I think CI need a kind label.

**Review by sanposhiho — COMMENTED** [review]
Thanks :)

/lgtm

**Review by alculquicondor — COMMENTED** [review]
nit

**Comment by tangwz** [comment]
@alculquicondor @sanposhiho PTAL

**Comment by sanposhiho** [comment]
please squash

**Review by sanposhiho — APPROVED** [review]
/lgtm

**Pull request merged**